### PR TITLE
feat(pipeline): Pipeline action initialization load

### PR DESCRIPTION
**Pull request by kakj-go**
#### What this PR does / why we need it:
Pipeline action initialization load

#### Which issue(s) this PR fixes:

- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb25JRHMiOlsxMTkwLDExNzQsMTIxOF0sImFzc2lnbmVlIjpbIjEwMDA1NjAiXX0%3D&id=307533&iterationID=1190&pId=306714&type=TASK)

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |        Pipeline action initialization load      |
| 🇨🇳 中文    |       pipeline action 初始化加载       |

### CODE REVIEW

File: modules/pipeline/providers/actionmgr/mock.go [hunk 1]
```diff
@@ -51,8 +51,8 @@ func (m *MockActionMgr) MakeActionLocationsBySource(source apistructs.PipelineSo
 		locations = append(locations, apistructs.PipelineTypeFDP.String()+"/")
 	case apistructs.PipelineSourceDice, apistructs.PipelineSourceProject, apistructs.PipelineSourceProjectLocal, apistructs.PipelineSourceOps, apistructs.PipelineSourceQA:
 		locations = append(locations, apistructs.PipelineTypeCICD.String()+"/")
-	default:
-		locations = append(locations, apistructs.PipelineTypeDefault.String()+"/")
 	}
+
+	locations = append(locations, apistructs.PipelineTypeDefault.String()+"/")
```
Comment (sfwn) [inline modules/pipeline/providers/actionmgr/mock.go:56]: just for mock use, do not add unit-test for this.

File: modules/pipeline/providers/actionmgr/mock_test.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+package actionmgr
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/erda-project/erda/apistructs"
+)
+
+func TestMockActionMgr_MakeActionLocationsBySource(t *testing.T) {
```
Comment (sfwn) [inline modules/pipeline/providers/actionmgr/mock_test.go:24]: delete it

File: modules/pipeline/providers/actionmgr/provider.go [hunk 3]
```diff
@@ -28,10 +28,13 @@ import (
 	"github.com/erda-project/erda/apistructs"
 	"github.com/erda-project/erda/bundle"
 	"github.com/erda-project/erda/modules/pipeline/providers/actionmgr/db"
+	"github.com/erda-project/erda/modules/pipeline/providers/edgepipeline_register"
 	"github.com/erda-project/erda/pkg/common/apis"
 	"github.com/erda-project/erda/pkg/goroutinepool"
 )
 
+const FilePath = "/app/extensions-init"
```
Comment (sfwn) [inline modules/pipeline/providers/actionmgr/provider.go:36]: make it configurable

Comment (sfwn) [reply modules/pipeline/providers/actionmgr/provider.go:36]: using p.Cfg

File: modules/pipeline/providers/actionmgr/provider.go [hunk 4]
```diff
@@ -62,6 +66,15 @@ func (s *provider) Init(ctx servicehub.Context) error {
 	s.defaultActionsCache = make(map[string]apistructs.ExtensionVersion)
 	s.pools = goroutinepool.New(s.Cfg.PoolSize)
 	s.bdl = bundle.New(bundle.WithAllAvailableClients())
+	go func() {
+		if s.EdgeRegister.IsEdge() {
+			return
+		}
+		err := s.actionService.InitAction(FilePath)
+		if err != nil {
+			s.Log.Errorf("failed to InitExtension error %v", err)
```
Comment (sfwn) [inline modules/pipeline/providers/actionmgr/provider.go:75]: what will happen if InitExtension failed? does one pipeline can start&run correctly at edge side?

File: apistructs/extention.go [hunk 5]
```diff
@@ -26,6 +26,11 @@ import (
 
 const DicehubExtensionsMenu = "dicehub.extensions.menu"
 
+type SpecType string
```
Comment (sfwn) [inline apistructs/extention.go:29]: what is `SpecType`, just use `ExtensionType`.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/pipeline/providers/actionmgr/provider.go (hunk 4 above)
Comment (kakj-go) [reply modules/pipeline/providers/actionmgr/provider.go:75]: Synchronization is from the center to the edge. Only the center will start the InitAction, and then call the save method to synchronize to the edge

Comment (kakj-go) [reply modules/pipeline/providers/actionmgr/provider.go:75]: I changed the InitAction method. Only a single action synchronization will fail, and the overall InitAction method will not fail

**Review by sfwn — APPROVED** [review]

**Pull request merged**